### ISSUE TITLE
Training loss is not converging after 1 epoch Description

### ISSUE THREAD

**Issue by yimuu** [jcwang0602/MLLMSeg#4]
**1. Describe the bug**
During training, the loss function does not show a clear trend of convergence. The loss values fluctuate significantly throughout the first epoch, and the Intersection over Union (IoU) metric remains very low, indicating that the model is not learning effectively. The training process seems to stall at the end of the first epoch with the learning rate dropping to 0.

**2. To Reproduce**
1. Start the training script with the current configuration.
2. Monitor the training logs from the beginning.
3. Observe the loss values and IoU metric per iteration.
4. Notice that the loss fluctuates and does not decrease consistently as the epoch progresses.

**3. Expected behavior**
I expect to see the total loss (`loss`) and its components (`ce_loss`, `mask_loss`) steadily decrease over the course of the training epoch. Correspondingly, the `iou` metric should show a consistent increase, indicating that the model's performance is improving.

**4. Actual behavior**
The loss values are erratic and show no clear downward trend. The `iou` metric is stagnant at a very low value (around 0.03 - 0.08). The training concludes the first epoch by setting the `learning_rate` to `0.0`, which suggests the learning rate scheduler might be completing prematurely or there's an issue with the training loop.

**5. Training Logs Snippets**
Here are some representative log entries from the training process:

*   **Start of Epoch 1:**
    ```json
    {'loss': 2.8536, 'ce_loss': 1.9031, 'mask_loss': 0.9505, 'iou': 0.0373, 'epoch': 1.0}
    ```

*   **Middle of Epoch 1:**
    ```json
    {'loss': 3.0870, 'ce_loss': 2.1451, 'mask_loss': 0.942, 'iou': 0.0354, 'epoch': 1.0}
    ```
    ```json
    {'loss': 2.8779, 'ce_loss': 1.9754, 'mask_loss': 0.9026, 'iou': 0.0585, 'epoch': 1.0}
    ```

*   **End of Epoch 1:**
    ```json
    {'loss': 2.8819, 'ce_loss': 1.9926, 'mask_loss': 0.8893, 'iou': 0.0813, 'epoch': 1.0}
    ```
    ```json
    {'loss': 2.7775, 'ce_loss': 1.9019, 'mask_loss': 0.8756, 'iou': 0.0774, 'epoch': 1.0}
    ```

*   **Final Log Entry:**
    ```json
    {'loss': 2.8002, 'learning_rate': 0.0, 'epoch': 1.0}
    ```

Any help or suggestions on how to debug this non-converging behavior would be greatly appreciated. Thank you

**Comment by jcwang0602** [comment]
Could you provide your pytorch and cuda versions?

**Comment by jcwang0602** [comment]
I am trying to retrain now and have completed 1500 steps so far, which seems to be normal.

<img width="2486" height="1096" alt="Image" src="https://github.com/user-attachments/assets/4bd77b54-dbe8-495a-8580-c7af003828ec" />

According to our experience, if other versions of PyTorch and CUDA are used, the model may not converge.
